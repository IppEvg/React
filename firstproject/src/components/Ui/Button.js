import styles from '../func/message.module.scss'

export function Button (props){
    return(
        <>
        <button className={styles.button}  {...props} onClick={props.click} >{props.children}</button>
        </>
    )
}

