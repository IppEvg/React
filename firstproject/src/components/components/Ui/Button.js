import styles from '../func/message.module.scss'

export function Button (props){
    return(
        <>
        <button className={styles.button} {...props} >{props.children}</button>
        </>
    )
}

