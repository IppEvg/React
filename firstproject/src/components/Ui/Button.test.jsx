import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

import {Button} from './Button.js'

describe ('testing button',()=>{
    it('render component Button', () => {
        render(<Button>Change name</Button>)
      })

    it ("render button with snapshot", ()=>{
    const {asFragment} = render(<Button>Send</Button>)
    expect (asFragment()).toMatchSnapshot()
})
    it('button click with userEvent', async () => {
    const mockHandler = jest.fn();

    render (<Button click={mockHandler}>Sign in</Button>)
    await userEvent.click(screen.getByText(/Sign in/))
    expect(mockHandler).toHaveBeenCalledTimes(1)
  })
// it ('button click with userEvent',async ()=>{
//     const mockHandler = jest.fn()
//     render (<Button onClick={mockHandler}>Delete</Button>)
//     await userEvent.onClick(screen.getByText(/Delete/))
//     expect(mockHandler).toHaveBeenCalledTimes(1)
//  })
})
