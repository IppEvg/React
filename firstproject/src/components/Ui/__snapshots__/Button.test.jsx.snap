// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing button render button with snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Send
  </button>
</DocumentFragment>
`;
